__init__.py

```diff
@@ -662,6 +691,143 @@ def _get_alarm_matches(
+        except:
+            self.log.debug("Not a roman numeral")
+
+        if dom is None:
+            dom = extract_number(utt)
+
+        # if the user says a day of the week (mon-sun) we 
+        # keep them in a separate list of day of week matches
+        dow_matches = []
+
+        # we also look for specific time matches like 9am
+        tod_matches = []
+        if when_utc:
+            tod_matches = self.get_tod_matches(when.time(), alarm_list)
+
+        # these alarms match our when_utc date only
+        date_matches = []
+
+        # these alarms match any numbers passed in like
+        # the 25th, 5 etc which we assume are dom
+        # hopefully this won't muss up ordinals
+        dom_matches = []
+
+        for alm in alarm_list:
+            self.log.debug("    Loop: %s" % (alm,))
+
+            # get utc time from alarm timestamp
+            dt_obj = datetime.fromtimestamp( alm['timestamp'] )
+            dt_obj = dt_obj.astimezone(pytz.utc)
+
+            # we also need a local version of our utc time
+            cfg_tz = pytz.timezone(self.local_tz) 
+            dt_local = dt_obj.astimezone(cfg_tz)
+
+            if user_dow == dt_local.weekday():
+                dow_matches.append(alm)
+
+            if dom == dt_local.day:
+                dom_matches.append(alm)
+
+            if have_time:
+                # unambiguous
+                if when_utc == dt_obj:
+                    exact_matches.append(alm)
+            else:
+                self.log.debug("    Loop: when:%s, dt_obj:%s" % (when_utc, dt_obj))
+                if when_utc is not None and dt_obj is not None:    
+                    # do dates match ?
+                    if when_utc.strftime("%y-%m-%d") == dt_obj.strftime("%y-%m-%d"):
+                        date_matches.append(alm)
+
+        if len(exact_matches) > 0:
+            return exact_matches
+
+        if len(date_matches) > 0:
+            return date_matches
+
+        if len(tod_matches) > 0:
+            return tod_matches
+
+        if len(dow_matches) > 0:
+            return dow_matches
+
+        return dom_matches
+
+    def get_tod_matches(self, alarm_time, alarm_list):
+        ''' find alarms where the time matches'''
+        tod_matches = []
+
+        for alarm in alarm_list:
+            # get utc time from alarm timestamp
+            dt_obj = datetime.fromtimestamp( alarm['timestamp'] )
+            dt_obj = dt_obj.astimezone(pytz.utc)
+
+            # we also need a local version of our utc time
+            cfg_tz = pytz.timezone(self.local_tz) 
+            dt_local = dt_obj.astimezone(cfg_tz)
+
+            self.log.debug("Check for TOD matches, dt_local.time()=%s" % (dt_local.time(),))
+            if dt_local.time() == alarm_time:
+                tod_matches.append(alarm)
+
+        return tod_matches
+
+    def get_dow_from_utterance(self, utt):
+        user_dow = None
+        result = re.findall('(mon|tues|wed|thurs|fri|sat|sun)day', utt)
```

Comment:
can we use `self.weekdays` here instead so it's localized?

Reply:
I'm gonna leave the last two to you. I believe ordinals are indices not days of the month as I mentioned in a previous comment and the above snippet has nuances regarding the minimized day of week name (notice they are not monday, tuesday, etc) and how the data arrives so please feel free to make any stylistic changes you would like.